File: test/data/cpp/test06/StateState2.cpp
#include "StateState2.h"
StateState2::StateState2(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState2::~StateState2()
{
}
BaseState* StateState2::TransitAborted()
{
    return 0;
}
BaseState* StateState2::TransitDeepResume()
{
    _stateController->InstanceOfState3->Setup(true, true);
    return _stateController->InstanceOfState3;
}
BaseState* StateState2::TransitResume()
{
    _stateController->InstanceOfState3->Setup(true, false);
    return _stateController->InstanceOfState3;
}
BaseState* StateState2::TransitSucceeded()
{
    _stateController->InstanceOfState3->Setup(false, false);
    return _stateController->InstanceOfState3;
}
const char* StateState2::GetStateName()
{
    return "State2";
}
BaseState* StateState2::GetParent()
{
    return 0;
}
int StateState2::GetStateID()
{
    return 2;
}

File: test/data/typescript/test06/StateState2.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateState2 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitAborted(): BaseState | null
    {
        return null;
    }
    public TransitDeepResume(): BaseState | null
    {
        this._stateController.InstanceOfState3.Setup(true, true);
        return this._stateController.InstanceOfState3;
    }
    public TransitResume(): BaseState | null
    {
        this._stateController.InstanceOfState3.Setup(true, false);
        return this._stateController.InstanceOfState3;
    }
    public TransitSucceeded(): BaseState | null
    {
        this._stateController.InstanceOfState3.Setup(false, false);
        return this._stateController.InstanceOfState3;
    }
    public  GetStateName(): string
    {
        return "State2";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
    public GetStateID(): number
    {
        return 2;
    }
}